src/analyze/gui/tests/tst_middle_elide.cpp

```diff
@@ -0,0 +1,64 @@
+#include <QtTest>
+#include <QString>
+#include "analyze/gui/middleelide.h"
+
+// add necessary includes here
+
+const char* simple_case = "MainWindow::onLoadingFinish(unsigned int&)";
+const char* one_bracket = "std::vector<test type in bracket> MainWindow::onLoadingFinish(unsigned int&)";
+const char* one_bracket_fixed = "std::vector<...> MainWindow::onLoadingFinish(unsigned int&)";
+const char* two_brackets = "std::vector<test type in bracket> MainWindow<vector_a>::onLoadingFinish(unsigned int&)";
+const char* two_brackets_fixed = "std::vector<...> MainWindow<...>::onLoadingFinish(unsigned int&)";
+const char* nested_brackets = "std::vector<test type <int> in bracket> MainWindow::onLoadingFinish(unsigned int&)";
+const char* nested_brackets_fixed = "std::vector<...> MainWindow::onLoadingFinish(unsigned int&)";
+
+
+class test_initilaize : public QObject
+{
+    Q_OBJECT
+
+public:
+    test_initilaize() = default;
+    ~test_initilaize() = default;
+
+private slots:
+    void test_simple_case();
```

Comment:
use data driven tests instead (`QTest::add{Column,Row}` & friends)

Reply:
I am familiar with the data-driven tests of the QtTest framework.
However, I wanted to name the individual tests explicitly.
For me, it is much clearer to have individual tests that precisely describe the individual test case.
(e.g. test_single_bracket(), test_multiple_brackets()).

This approach (originally from TDD) describes the tests much more clearly than a data-driven approach with only one method.
What would still be possible in any case is a summary of the redundant code of the individual test methods.

Reply:
I don't follow, the rows get a label which would have the same explanation, no? i.e. to me, the following has the same level of detail:

```
testElideTemplates:single_bracket

testElideTemplates_singleBracket
```